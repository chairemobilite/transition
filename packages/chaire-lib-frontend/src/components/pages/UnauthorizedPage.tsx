/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import { Link } from 'react-router';
import { useTranslation } from 'react-i18next';

export const UnauthorizedPage: React.FC = () => {
    const { t } = useTranslation('auth');

    return (
        <div>
            {t('auth:Unauthorized')} - <Link to="/login">{t('auth:BackToLoginPage')}</Link>
        </div>
    );
};

export default UnauthorizedPage;
