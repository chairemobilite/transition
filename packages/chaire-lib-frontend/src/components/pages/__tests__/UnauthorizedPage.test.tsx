/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import { BrowserRouter } from 'react-router';
import { render } from '@testing-library/react';
import UnauthorizedPage from '../UnauthorizedPage';

test('Should correctly render Unauthorized page', () =>{
    const { container } = render(<BrowserRouter><UnauthorizedPage /></BrowserRouter>);
    expect(container).toMatchSnapshot();
});
