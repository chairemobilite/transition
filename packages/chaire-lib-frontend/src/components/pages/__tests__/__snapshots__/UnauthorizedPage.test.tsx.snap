// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Unauthorized page 1`] = `
<BrowserRouter>
  <Router
    history={
      {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withI18nextTranslation(UnauthorizedPage)>
      <UnauthorizedPage
        i18n={{}}
        t={[Function]}
        tReady={false}
      >
        <div>
          Unauthorized
           - 
          <Link
            to="/login"
          >
            <LinkAnchor
              href="/login"
              navigate={[Function]}
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                BackToLoginPage
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </UnauthorizedPage>
    </withI18nextTranslation(UnauthorizedPage)>
  </Router>
</BrowserRouter>
`;
